scripts/app/adMover.js

```diff
@@ -0,0 +1,73 @@
+window.onload = function () {
+    "use strict";
+    let ksfAdMover = {};
+    ksfAdMover.checkPara = function (ptag, expectedType, expectedClass) {
+        let response = false;
+        expectedType = expectedType.toUpperCase(); // nodeName is always uppercase
+        // check that the surrounding paragraphs are OK when placing the in text ad
+        let pBefore = ptag.previousElementSibling;
+        let pBeforeContent = pBefore.innerHTML.replace(/[^A-Za-z0-9]/g, '');
+        let pBeforeClass = pBefore.classList.contains(expectedClass);
+        let pSelf = ptag;
+        let pSelfContent = pSelf.innerHTML.replace(/[^A-Za-z0-9]/g, '');
+        let pSelfClass = pSelf.classList.contains(expectedClass);
+        if ((pBeforeContent !== "" && pSelfContent !== "" && !(/^\s*$/.test(pBeforeContent)) && !(/^\s*$/.test(pSelfContent))) && (pSelf.nodeName === expectedType && pBefore.nodeName === expectedType) && (pBeforeClass && pSelfClass)) {
```

Comment:
I think the `!(/^\s*$/.test(pBeforeContent))` will also cover the `pBeforeContent !== ""` case, so the latter can be removed.

Reply:
You are right. I do think that in the case of an empty string the first test i a lot more performant. But in practice that is probably not significant as this at worst will be run like two times. Also that line is quite ugly so cleaning it up a bit is a value in itself.